### ISSUE TITLE
PEP 3105 Make print a function in docs/core/howto/listings/ and docs/core/howto/tutorial/listings/

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8410]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8410|
|Type|enhancement|
|Created|2016-06-03 21:18:45Z|
|Branch|https://github.com/twisted/twisted/tree/8410-rodrigc-print|

This ticket was split out of [ticket:5812] which was too big to review.

PEP 3105 ( ​​​https://www.python.org/dev/peps/pep-3105/ ) **print** deprecates has changed from a statement to a function.

In Python 2.6 and Python 2.7, the default is print as a statement

```
print "Hello"
```

In Python 3.x, the above example is an error, since only print as a function is available.

To consistently use **print** as a function
on Python 2.6, Python 2.7, and Python 3.x, this should be done:

```
from __future__ import print_function

print("Hello")
```

<details><summary>Searchable metadata</summary>

```
trac-id__8410 8410
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8410_rodrigc_print 8410-rodrigc-print
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464988725432394 1464988725432394
changetime__1466013912185187 1466013912185187
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/126

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Changes look good.

Some files have a print import even if they don't include a print function, but I think that is good to have this for the future as I hope that this will remind us to use py3 compatible code.

Now, since we don't have test for the documentation, we might end up with broken examples ... but this is another story.

Waiting for build results and will merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set status to `closed`|
|-|-|

In changeset 6a80dc65e9bc963b2a1372d29b716adbb3983713
```
#!CommitTicketReference repository="" revision="6a80dc65e9bc963b2a1372d29b716adbb3983713"
Merge 8410-rodrigc-print: Use print function in docs/core/howto.

Author: rodrigc
Reviewer: adiroiban
Fixes: #8410
```
